sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/Transforms.java

```diff
@@ -315,10 +316,17 @@ static PathProperties toPathProperties(BlobProperties properties) {
                 Transforms.toDataLakeArchiveStatus(properties.getArchiveStatus()), properties.getEncryptionKeySha256(),
                 properties.getAccessTierChangeTime(), properties.getMetadata(), properties.getExpiresOn());
 
-            return AccessorUtility.getPathPropertiesAccessor().setPathProperties(pathProperties, properties.getEncryptionScope());
+            return AccessorUtility.getPathPropertiesAccessor().setPathProperties(pathProperties, properties.getEncryptionScope(), encryptionContext);
         }
     }
 
+    static String getEncryptionContext(Response<?> r) {
+        if (r == null) {
+            return null;
+        }
+        return r.getHeaders().getValue("x-ms-encryption-context");
```

Comment:
nit: Create an `HttpHeaderName` for `x-ms-encryption-context` and use the `HttpHeaderName` API instead, it performs much much better.

Reply:
Using a string Constant for x-ms-encryption-context to avoid dependency on azure-core package.

Reply:
Oh, you can just create a local, or shared constant, anywhere you want and it'll be more performant than using a String constant. So instead of the following in `Constants`:

`public static final String X_MS_ENCRYPTION_CONTEXT = "x-ms-encryption-context";`

you can do

`public static final HttpHeaderName X_MS_ENCRYPTION_CONTEXT = HttpHeaderName.fromString("x-ms-encryption-context");`